pkg/analysis/optionalfields/analyzer.go

```diff
@@ -190,7 +210,40 @@ func (a *analyzer) checkFieldPropertiesWithoutOmitEmpty(pass *analysis.Pass, fie
 		// Once it has the omitempty tag, it will also need to be a pointer in some cases.
 		// Now handle it as if it had the omitempty already.
 		// We already handle the omitempty tag above, so force the `hasOmitEmpty` to true.
-		a.checkFieldPropertiesWithOmitEmptyRequired(pass, field, fieldName, jsonTags, underlying, true, hasValidZeroValue, completeValidation, isPointer, isStruct)
+		a.checkFieldPropertiesWithOmitEmptyRequired(pass, field, fieldName, jsonTags, underlying, true, false, hasValidZeroValue, completeValidation, isPointer, isStruct)
+	}
+}
+
+func (a *analyzer) checkFieldPropertiesWithOmitZeroRequired(pass *analysis.Pass, field *ast.Field, fieldName string, jsonTags extractjsontags.FieldTagInfo, hasOmitZero, hasValidZeroValue, isPointer, isStruct bool) {
+	// In this case, we should always add the omitzero for struct fields if it isn't present.
+	a.handleFieldShouldHaveOmitZero(pass, field, fieldName, hasOmitZero, isStruct, jsonTags)
```

Comment:
Don't we need to check that it's a struct before we suggest the omitzero? To my knowledge it's only structs we are targeting with this change for now

In the case that the zero value is valid, and we do think we want a `pointer` on the struct, does `omitzero` make sense?

Reply:
> Don't we need to check that it's a struct before we suggest the omitzero? To my knowledge it's only structs we are targeting with this change for now

I have done that check inside the `handleFieldShouldHaveOmitZero` with
```
	if hasOmitZero || !isStruct {
		return
	}
```

> In the case that the zero value is valid, and we do think we want a pointer on the struct, does omitzero make sense?

Even I had this question and thought of asking, but as per our earlier conversation, We need to add both omitempty and omitzero right.

Reply:
I think if there's a pointer, you don't need `omitzero`, but should have `omitempty`, if there isn't a pointer, we should do `omitzero` and `omitempty`

Reply:
sure, thanks, Then I think we can remove pointer and make sure to have omitzero tag.